integrations/src/test/java/io/jenkins/plugins/casc/ViewJobFiltersTest.java

```diff
@@ -0,0 +1,55 @@
+package io.jenkins.plugins.casc;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+import java.util.List;
+
+import hudson.views.BuildDurationFilter;
+import hudson.views.BuildStatusFilter;
+import hudson.views.SecurityFilter;
+
+import org.jfrog.hudson.ArtifactoryBuilder;
+import org.jfrog.hudson.ArtifactoryServer;
+import org.junit.Rule;
+import org.junit.Test;
+
+import io.jenkins.plugins.casc.misc.ConfiguredWithReadme;
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithReadmeRule;
+import jenkins.model.Jenkins;
+
+public class ViewJobFiltersTest {
+	
+	@Rule
+    public JenkinsConfiguredWithReadmeRule j = new JenkinsConfiguredWithReadmeRule();
+	
+	@Test
+    @ConfiguredWithReadme(value = "view-job-filters/README.md")
+    public void configure_view_job_filters() throws Exception {
+		
+		final BuildDurationFilter buildDurationFilter = new BuilDurationFilter();
+		assertTrue(buildDurationFilter.isLessThan());
+		assertThat(buildDurationFilter.getBuildDurationMinutes, is(5));
+		assertThat(buildDurationFilter.getAmount(), is(60.0));
+		assertThat(buildDurationFilter.getAmountTypeString(), is("Days"));
+		assertThat(buildDurationFilter.getBuildCountTypeString(), is("Latest"));
+		assertTrue(buildDurationFilter.isIncludeMatched());	
+		
```

Comment:
Hi @timja, I referred to other test files to try to understand how to write tests, however, can you tell me if this is the right way to test my yaml configuration for this plugin?

Now, that I correctly imported the classes (by specifying the dependencies), I see that constructor BuildDurationFilter() is not defined. There's only one constructor and that is parameterized with all the above-mentioned values.

So I'll pass these hardcoded values into the constructor, then use the getters to fetch these values from the object and test it against what? I guess I'll test them against the values in the yaml file. If so, how do I fetch them here? using the JenkinsConfiguredWithReadmeRule object?

Reply:
you need to retrieve the view, not create a filter with the constructor

Jenkins.get().getViews() or something like that and work your way down till you find the view